infrastructure/iac/iac_html.go

```diff
@@ -0,0 +1,94 @@
+/*
+ * © 2024 Snyk Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package iac
+
+import (
+	"bytes"
+	"crypto/rand"
+	_ "embed"
+	"encoding/base64"
+	"fmt"
+	"html/template"
+
+	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/domain/snyk"
+	iac "github.com/snyk/snyk-ls/infrastructure/iac/template/assets"
+	"github.com/snyk/snyk-ls/internal/product"
+)
+
+type TemplateData struct {
+	Styles       template.CSS
+	Issue        snyk.Issue
+	SeverityIcon template.HTML
+	Nonce        string
+}
+
+//go:embed template/index.html
+var detailsHtmlTemplate string
+
+//go:embed template/styles.css
+var stylesCSS string
```

Comment:
🙌  Directive to be able to inject the CSS form the CSS file so we don't write the rules in the HTML